Source/WebKit/WebProcess/WebPage/WebCookieJar.cpp

```diff
@@ -310,4 +323,33 @@ void WebCookieJar::setCookieAsync(WebCore::Document& document, const URL& url, c
     WebProcess::singleton().ensureNetworkProcessConnection().connection().sendWithAsyncReply(Messages::NetworkConnectionToWebProcess::SetCookieFromDOMAsync(document.firstPartyForCookies(), sameSiteInfo, url, frameID, pageID, applyTrackingPreventionInNetworkProcess, shouldRelaxThirdPartyCookieBlocking(webFrame), cookie), WTFMove(completionHandler));
 }
 
+#if HAVE(COOKIE_CHANGE_LISTENER_API)
+void WebCookieJar::addChangeListener(const String& host, const WebCore::CookieChangeListener& cookieChangeListener)
+{
+    auto it = m_changeListeners.find(host);
+    if (it != m_changeListeners.end() && it->value.contains(cookieChangeListener))
```

Comment:
This is *still* not correct. You're now avoiding sending the subscribe IPC when this listener is already registered. However, you're *still* unnecessarily sending the subscribe IPC if there is already a listener for this host.

Reply:
Should be fixed now!